gpcontrib/arenadata_toolkit/sql/upgrade_test.sql

```diff
@@ -0,0 +1,175 @@
+-- which were added at version 1.4
+	PERFORM arenadata_toolkit.adb_create_tables();
+
+	IF EXISTS (SELECT 1
+			   FROM information_schema.columns
+			   WHERE table_schema='arenadata_toolkit' AND
+			         table_name='db_files_current' AND
+			         column_name='tablespace_location')
+	THEN
+		RETURN NEXT from_version || ': tablespace_location at arenadata_toolkit.db_files_current check';
+	END IF;
+	IF EXISTS (SELECT 1
+			   FROM information_schema.columns
+			   WHERE table_schema='arenadata_toolkit' AND
+			         table_name='__db_files_current' AND
+			         column_name='tablespace_location')
+	THEN
+		RETURN NEXT from_version || ': tablespace_location at arenadata_toolkit.__db_files_current check';
+	END IF;
+	IF EXISTS (SELECT 1
+				   FROM information_schema.columns
+				   WHERE table_schema='arenadata_toolkit' AND
+				         table_name='__db_files_current_unmapped' AND
+				         column_name='tablespace_location')
+	THEN
+		RETURN NEXT from_version || ': tablespace_location at arenadata_toolkit.__db_files_current_unmapped check';
+	END IF;
+	IF EXISTS (SELECT 1
+			   FROM information_schema.columns
+			   WHERE table_schema='arenadata_toolkit' AND
+			         table_name='db_files_history' AND
+			         column_name='tablespace_location')
+	THEN
+		RETURN NEXT from_version || ': tablespace_location at arenadata_toolkit.db_files_history check';
+	END IF;
+	IF EXISTS (SELECT 1
+			   FROM pg_tables
+			   WHERE schemaname='arenadata_toolkit' AND
+			         tablename like 'db_files_history_backup%')
+	THEN
+		RETURN NEXT from_version || ': db_files_history_backup check';
+	END IF;
+
+-- Cleanup before next step
+	DROP EXTENSION arenadata_toolkit;
+
+-- Check create extension with the latest version after current was installed and dropped
+	CREATE EXTENSION arenadata_toolkit;
+	PERFORM arenadata_toolkit.adb_create_tables();
+	PERFORM arenadata_toolkit.adb_collect_table_stats();
+
+-- Check the result
+	IF EXISTS (SELECT 1
+			   FROM pg_available_extensions
+			   WHERE name='arenadata_toolkit' AND
+			         default_version=installed_version)
+	THEN
+		RETURN NEXT from_version || ': create the latest check';
+	END IF;
+
+-- Cleanup
+	DROP EXTENSION arenadata_toolkit;
+	DROP SCHEMA arenadata_toolkit CASCADE;
+
+END$$
+LANGUAGE plpgsql;
+
+CREATE FUNCTION do_test_for_each_versions()
+RETURNS setof TEXT
+AS $$
+DECLARE
+	version RECORD;
+BEGIN
+	FOR version IN
+		SELECT *
+		FROM (VALUES ('1.0'),('1.1'),('1.2'),('1.3'))
+			AS from_versions
+	LOOP
+		RETURN QUERY SELECT do_upgrade_test_for_arenadata_toolkit(
+								version.column1::TEXT,
+								(SELECT default_version
+								 FROM pg_available_extensions
+								 WHERE name = 'arenadata_toolkit'))
+					 ORDER BY 1;
+	END LOOP;
+END$$
+LANGUAGE plpgsql;
+
+SELECT do_test_for_each_versions() ORDER BY 1;
+
+-- Cleanup
+DROP FUNCTION do_test_for_each_versions();
+DROP FUNCTION do_upgrade_test_for_arenadata_toolkit(TEXT, TEXT);
```

Comment:
I still don't see a test for the bundled version of the extension.

Reply:
Bundle has three tables:
```
adb=# SELECT c.relname
FROM pg_class c
LEFT JOIN pg_namespace n ON n.oid = c.relnamespace
WHERE n.nspname = 'arenadata_toolkit';
               relname               
-------------------------------------
 db_files_history
 db_files_history_1_prt_default_part
 db_files_history_1_prt_p202312
 daily_operation
 operation_exclude
(5 rows)
```
These tables is created at function `mock_manual_installation()` at test `arenadata_toolkit_test`. This test checks how last version of arenadata_toolkit will be installed to bundle.